src/world/action/super_boots.c

```diff
@@ -1,5 +1,206 @@
+                if (collisionStatus->currentCeiling < 0) {
+                    playerStatus->position.y += playerStatus->gravityIntegrator[0];
+                }
+            }
+            if (--playerStatus->framesOnGround <= 0) {
+                playerStatus->fallState++;
+            }
+            break;
+        case 3:
+            playerStatus->position.y = player_check_collision_below(0.0f, &sp10);
+            D_802B6730 = 45.0f;
+            playerStatus->unk_8C += 45.0f;
+            if (playerStatus->unk_8C >= 360.0) {
+                playerStatus->unk_8C = 0.0f;
+                playerStatus->fallState++;
+                playerStatus->gravityIntegrator[1] = -3.4744f;
+                playerStatus->gravityIntegrator[2] = 0.4704f;
+                playerStatus->gravityIntegrator[3] = -1.1904f;
+                playerStatus->gravityIntegrator[0] = playerStatus->gravityIntegrator[1];
+                suggest_player_anim_clearUnkFlag(0x1000B);
+            }
+            break;
+        case 4:
+            temp_f20 = func_800E34D8();
+            playerStatus->position.y = player_check_collision_below(temp_f20, &sp10);
+            if (temp_f20 < -100.0f) {
+                playerStatus->gravityIntegrator[3] = 0.0f;
+                playerStatus->gravityIntegrator[2] = 0.0f;
+                playerStatus->gravityIntegrator[1] = 0.0f;
+                playerStatus->gravityIntegrator[0] = -100.0f;
+            }
+            if (sp10 >= 0) {
+                if (collisionStatus->currentFloor & 0x4000 && (entityType = get_entity_type(collisionStatus->currentFloor),
+                                                               entityType == 8 || entityType == 7)) {
+                    get_entity_by_index(collisionStatus->currentFloor)->collisionFlags |= 1;
+                    playerStatus->fallState = 0xB;
+                    playerStatus->flags &= ~0x8;
+                    break;
+                } else {
+                    colliderType = get_collider_type_by_id(sp10);
+                    if (colliderType == 3) {
+                        playerStatus->unk_BF = 1;
+                        set_action_state(ACTION_STATE_HIT_LAVA);
+                        playerStatus->flags |= 0x800;
+                        playerStatus->flags &= ~0x8;
+                        return;
+                    } else if (colliderType == 2) {
+                            set_action_state(ACTION_STATE_HIT_LAVA);
+                            playerStatus->flags &= ~0x8;
+                            return;
+                    } else {
+                        playerStatus->gravityIntegrator[1] = -3.4744f;
+                        playerStatus->gravityIntegrator[2] = 0.4704f;
+                        playerStatus->gravityIntegrator[3] = -1.1904f;
+                        playerStatus->gravityIntegrator[0] = 10.0f;
+                        suggest_player_anim_clearUnkFlag(0x1000B);
+                        playerStatus->actionState = 0xE;
```

Comment:
I think we should have an action state enum value for E - if not, could you add it?